service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -282,13 +253,39 @@ private String prepareStatementsForDerivedObservations(IdMapper idMapper, List<O
                     )
                 );
             } else {
-                derivedObservationsBlock.append(observationStatement).append(narrativeStatement);
+                derivedObservationsBlock.append(observationStatement).append(narrativeStatements);
             }
         });
 
         return derivedObservationsBlock.toString();
     }
 
+    private boolean observationDoesNotHaveCommentNoteCode(Observation observation) {
+        return observation.hasCode() && !CodeableConceptMappingUtils.hasCode(observation.getCode(), List.of(COMMENT_NOTE_CODE));
+    }
+
+    private String prepareCodeElement(Observation observation) {
+        if (observation.hasCode()) {
+            return codeableConceptCdMapper.mapCodeableConceptToCd(observation.getCode());
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private ClassCode prepareClassCode(List<Observation> derivedObservations) {
+        if (!derivedObservations.isEmpty()) {
+            Optional<Observation> derivedObservationWithoutCommentNoteCode = derivedObservations.stream()
+                    .filter(this::observationDoesNotHaveCommentNoteCode)
+                    .findFirst();
+
+            if (derivedObservationWithoutCommentNoteCode.isPresent()) {
+                return ClassCode.BATTERY;
+            }
+        }
+
+        return ClassCode.CLUSTER;
+    }
```

Comment:
You can still call `.stream()` on an empty collection; `.findFirst()` will necessarily return an empty optional.
```suggestion
    private ClassCode prepareClassCode(List<Observation> derivedObservations) {
        return derivedObservations.stream()
            .filter(this::observationDoesNotHaveCommentNoteCode)
            .map($ -> ClassCode.BATTERY) // if present map to BATTERY else still empty stream
            .findFirst() // convert to optional
            .orElse(ClassCode.CLUSTER); // if empty use CLUSTER instead
    }
```